server/client.go

```diff
@@ -3164,9 +3177,34 @@ func (c *client) processInboundClientMsg(msg []byte) bool {
 		didDeliver = c.sendMsgToGateways(c.acc, msg, c.pa.subject, c.pa.reply, qnames) || didDeliver
 	}
 
+	// Check to see if we did not deliver to anyone and the client has a reply subject set
+	// and wants notification of no_responders.
+	if !didDeliver && len(c.pa.reply) > 0 {
+		c.mu.Lock()
+		noRespondersRequested := c.opts.NoResponders
+		if noRespondersRequested {
```

Comment:
Just curious why not simply `if c.opts.NoResponders {`..

Reply:
yea.. Will fix, left over.

Reply:
Had locks different before.